ledger/src/builtins.rs

```diff
@@ -11,37 +11,33 @@ macro_rules! to_builtin {
 }
 
 /// Builtin programs that are always available
-fn genesis_builtins(cluster_type: ClusterType, bpf_jit: bool) -> Vec<Builtin> {
-    let mut builtins = vec![
+fn genesis_builtins(bpf_jit: bool) -> Vec<Builtin> {
+    vec![
         to_builtin!(solana_bpf_loader_deprecated_program!()),
         if bpf_jit {
             to_builtin!(solana_bpf_loader_program_with_jit!())
         } else {
             to_builtin!(solana_bpf_loader_program!())
         },
-    ];
-    if cluster_type != ClusterType::MainnetBeta {
-        builtins.push(if bpf_jit {
-            to_builtin!(solana_bpf_loader_upgradeable_program_with_jit!())
-        } else {
-            to_builtin!(solana_bpf_loader_upgradeable_program!())
-        });
-    }
-    builtins
+    ]
 }
 
 /// Builtin programs activated dynamically by feature
-fn feature_builtins() -> Vec<(Builtin, Pubkey, ActivationType)> {
+fn feature_builtins(bpf_jit: bool) -> Vec<(Builtin, Pubkey, ActivationType)> {
     vec![(
-        to_builtin!(solana_bpf_loader_upgradeable_program!()),
+        if bpf_jit {
```

Comment:
"with_jit" was missing for `feature_builtins()` activation of the upgradeable loader too